### PR TITLE
nms -> nms.py, added hessian detector

### PR DESCRIPTION
**Pull request by ducha-aiki**

### CODE REVIEW

**Comment by edgarriba** [comment]
@ducha-aiki could you rebase from master and add some tests? https://github.com/arraiyopensource/torchgeometry/pull/139 has been rebased. Feel free to divide the module by detectors/descriptors. However, where should we place nms ?

File: torchgeometry/feature/__init__.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 from .harris import CornerHarris, corner_harris
-from .harris import NonMaximaSuppression2d, non_maxima_suppression2d
+from .hessian import Hessian, hessian
+from .nms import NonMaximaSuppression2d, non_maxima_suppression2d
```
Comment (edgarriba) [inline torchgeometry/feature/__init__.py]: `from torchgeometry.feature.nms`

File: torchgeometry/feature/hessian.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+
+# functional api
+
+
+def hessian(input: torch.Tensor, scale: float = 1.0) -> torch.Tensor:
+    r"""Computes the first order image derivative in both x and y directions
+        using a Sobel operator.
+
+    See :class:`~torchgeometry.feature.CornerHarris` for details.
```
Comment (edgarriba) [inline torchgeometry/feature/hessian.py]: See :class:`~torchgeometry.feature.Hessian` for details.

File: torchgeometry/image/sobel.py [hunk 3]
```diff
@@ -112,6 +112,17 @@ def spatial_gradient(input: torch.Tensor) -> torch.Tensor:
     return SpatialGradient()(input)
 
 
+def spatial_gradient_2nd_order(input: torch.Tensor) -> torch.Tensor:
```
Comment (edgarriba) [inline torchgeometry/image/sobel.py]: we have laplace operator. Can we use it for this ?

Comment (ducha-aiki) [reply torchgeometry/image/sobel.py]: Not in current form: we need to have 1d gradients for gxx and gyy, while current laplacian is 2d

Comment (edgarriba) [reply torchgeometry/image/sobel.py]: ok, gotcha. Do you think there's any optimization to be applied here to bootstrap 2nd order ?

**Pull request closed without merging**